gis/openstreetmap-carto/Portfile

```diff
@@ -0,0 +1,203 @@
+license             public-domain
+
+description         A general-purpose OpenStreetMap mapnik style, in CartoCSS
+
+long_description    These are the CartoCSS map stylesheets for the Standard \
+                    map layer on OpenStreetMap.org.
+
+checksums           rmd160  f3effe425ea8c0d54f5437f22fe0181bb1005ea3 \
+                    sha256  be63d4fc06f1db447f1a249515f9fc83449f7f3027469d1cc051150ded3f85e9 \
+                    size    2719803
+
+python.default_version 39
+
+depends_lib-append  port:py${python.version}-psycopg2 \
+                    port:py${python.version}-requests \
+                    port:py${python.version}-yaml
+
+# Database variants (from the GDAL port)
+set postgresql_suffixes {12 13}
+
+set portsgresql_variants {}
+set postgresql_default_variant "if {"
+
+foreach s ${postgresql_suffixes} {
+    lappend portsgresql_variants postgresql${s}
+    set postgresql_default_variant "${postgresql_default_variant}!\[variant_isset postgresql${s}] && "
+}
+
+set postgresql_default_variant [string range ${postgresql_default_variant} 0 end-4]
+set postgresql_default_variant "${postgresql_default_variant}} {default_variants +postgresql${s}}"
+eval $postgresql_default_variant
+
+foreach s ${postgresql_suffixes} {
+    set p postgresql${s}
+    set v [string index ${s} 0].[string index ${s} 1]
+    set i [lsearch -exact ${portsgresql_variants} ${p}]
+    set c [lreplace ${portsgresql_variants} ${i} ${i}]
+    variant ${p} description "Enable PostgreSQL ${v} support" conflicts {*}${c} "
+        depends_lib-append      port:${p}
+    "
+}
+
+set datadir         ${prefix}/var/lib/${name}
+set configdir       ${prefix}/etc/${name}
+
+use_configure       no
+
+# The `scripts-get-external-data-py.diff` patch has been merged
+# into the upstream master at
+# https://github.com/gravitystorm/openstreetmap-carto/commit/4ec2dc9391c411e124c78b3ba1aad9173fea20cb
+# so it should be possible to remove it after the next release (> v5.3.1).
+patchfiles          style-fonts-mss.diff \
+                    scripts-get-external-data-py.diff
+
+variant postgresql12 conflicts postgresql13 description {Use with PostgreSQL 12} {
+    require_active_variants py${python.version}-psycopg2 postgresql12
+}
+
+variant postgresql13 conflicts postgresql12 description {Use with PostgreSQL 13} {
+    require_active_variants py${python.version}-psycopg2 postgresql13
+}
+
+post-patch {
+    reinplace "s#/usr/bin/env python3#${python.bin}#g" \
+        ${worksrcpath}/scripts/get-external-data.py \
+        ${worksrcpath}/scripts/generate_road_colours.py \
+        ${worksrcpath}/scripts/generate_shields.py \
+        ${worksrcpath}/scripts/indexes.py
+}
+
+build {}
+
+pre-destroot {
+    reinplace "s#data_dir: data#data_dir: ${datadir}#" ${worksrcpath}/external-data.yml
+}
+
+destroot {
+    xinstall -d ${destroot}${prefix}/etc/${name}
+    xinstall -m 0755 -o nobody -g admin -d ${destroot}${datadir}
```

Comment:
is it needed to change the owner of this directory?

Reply:
I'm not sure what is best here.  One of the scripts that this project includes, imports polygon map data describing shapes such as coastlines, ice-sheets and water boundaries.  I'd like to call this script from an all-encompassing startup and configuration shell script (delivered in the future mod_tile port), running as a non-root system user.  The data directory will need to be writeable by that user.

Perhaps the better alternative is not to create that directory at all here, but as part of the shell script and have the mod_tile port 'own' the directory.  On the other hand, the Use Case/Story kind of belongs to this port.  I'm more than happy to go along with your recommendation.  The location of the data directory is configureable, so can be handled by the shell script if we leave it's creation out of this port file.

Reply:
I've had a more in-depth look at this.  Whilst I can override the value entered in the `external-data.yml` configuration file in the mod_tile port's setup script, (by passing it as a parameter to the `get-external-data.py` script of this port,) it is nasty from an end-user perspective, because changing the value in the configuration file would have no effect.  It would be very confusing when it doesn't work, and discovering why would require debugging the scripts.  I could add a warning if the apparently configured YAML value differs, but that's a little ugly too.

Overall, I think it would be better to have this port create the data directory, writeable by the non-root system user that mod_tile setup script uses and have the mod_tile setup script respect the configuration path in `external-data.yml`.